tests/testthat/test-namespace_linter.R

```diff
@@ -18,6 +18,17 @@ test_that("namespace_linter respects check_exports and check_nonexports argument
   expect_lint("stats:::ssd(c(1,2,3))", NULL, namespace_linter(check_exports = FALSE, check_nonexports = FALSE))
 })
 
+test_that("namespace_linter respects check_exports and check_nonexports arguments", {
+  skip_if_not_installed("rlang")
+  linter <- namespace_linter()
+
+  expect_lint("rlang::`%||%`", NULL, linter)
+  expect_lint("rlang::`%||%`()", NULL, linter)
+
+  expect_lint("rlang::'%||%'", NULL, linter)
```

Comment:
Test `"%>%"` as well just in case?

Reply:
Won't we need to add `{magrittr}` to `Suggests` in that case? To make that namespace available?

Reply:
Why is that? We can use `rlang` for that as well. It's just a different way of quoting `%>%`.

Reply:
I am not sure what you mean: `%>%` is not exported from `{rlang}`.

Reply:
@AshesITR I've added a test, well sort of. See if it works for you.

Reply:
Now I'm completely confused 😅
Shouldn't the linter fire for non-exported objects, i.e. the tests fail?
I'll look at it in more detail later.

Reply:
This is the existing behaviour, and is unrelated to the current PR. Linter doesn't fail for non-exported objects, just produces a lint.

https://github.com/r-lib/lintr/blob/1ac48e94ef550945d94c0e47b43a9eb8041b5f00/tests/testthat/test-namespace_linter.R#L30-L34

Reply:
Ah, my bad. I just noticed `%||%` is tested, not `%>%`.
My original remark was supposed to be test `rlang::"%||%"` in addition to the other quoting options.
Sorry for the confusion!

Reply:
I see. So, in addition to `rlang::'%||%'`, which is already covered, you want me to check `rlang::"%||%"`.

Can do.